packages/typespec-azure-core/src/decorators.ts

```diff
@@ -350,7 +350,7 @@ export function extractLroStates(
 
         return diagnostics.wrap(undefined);
       } else {
-        storeLroState(program, result, option.value, variant);
+        storeLroState(program, result, variant.name.toString(), variant);
```

Comment:
Do we have to account for both enum and union types?

Reply:
yes that was the problem this broke the anonymous variant case. So in case of named variant change the variant name, in the case of unnamed check the value